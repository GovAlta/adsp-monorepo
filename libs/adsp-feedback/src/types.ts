export interface FeedbackContext {
  tenant?: string;
  site: string;
  view: string;
  correlationId?: string;
}

export interface FeedbackOptions {
  tenant?: string;
  name?: string;
  email?: string;
  site?: string;
  apiUrl?: string;
  getAccessToken?: () => Promise<string>;
  getContext?: () => Promise<FeedbackContext>;
}

export interface AdspFeedback {
  initialize(options: FeedbackOptions): void;
}

declare global {
  // eslint-disable-next-line no-var
  var adspFeedback: AdspFeedback;
}
