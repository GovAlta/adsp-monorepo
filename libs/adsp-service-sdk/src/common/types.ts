export enum SecurityClassifications {
  Protected_A = 'Protected A',
  Protected_B = 'Protected B',
  Protected_C = 'Protected C',
  Public = 'Public',
}
