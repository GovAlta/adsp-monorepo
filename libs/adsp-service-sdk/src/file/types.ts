export interface Rule {
  createdAt: string;
  active: true;
}

export interface Retention extends Rule {
  deleteInDays: number;
}

export interface FileTypeRules {
  retention?: Retention;
}
export interface FileType {
  id: string;
  name: string;
  anonymousRead: boolean;
  readRoles: string[];
  updateRoles: string[];
  rules?: FileTypeRules;
  securityClassification?: string;
}

export enum SecurityClassifications {
  Protected_A = 'Protected A',
  Protected_B = 'Protected B',
  Protected_C = 'Protected C',
  Public = 'Public',
}
