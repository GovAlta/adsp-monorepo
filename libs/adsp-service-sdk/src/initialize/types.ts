import { RequestHandler } from 'express';
import { Strategy } from 'passport';
import type { Logger } from 'winston';
import { TokenProvider } from '../access';
import { CombineConfiguration, ConfigurationConverter, ConfigurationService } from '../configuration';
import { ServiceDirectory } from '../directory';
import { EventService } from '../event';
import { PlatformHealthCheck } from '../healthCheck';
import { ServiceRegistration } from '../registration';
import { TenantService } from '../tenant';
import { AdspId } from '../utils';

export interface LogOptions {
  logger?: Logger;
  logLevel?: string;
}

export interface PlatformOptions extends ServiceRegistration {
  /**
   * Realm: Tenant realm of the service.
   *
   * The realm determines OIDC provider the service will use to verify and request access tokens.
   *
   * @type {string}
   * @memberof PlatformOptions
   */
  realm: string;
  /**
   * Client secret: Client secret of the service.
   *
   * This credential is used to retrieve access tokens for the service account of this service.
   *
   * @type {string}
   * @memberof PlatformOptions
   */
  clientSecret: string;
  /**
   * Directory URL: URL to the Directory of Services.
   *
   * The directory is used to look up other service and API URLs.
   *
   * @type {URL}
   * @memberof PlatformOptions
   */
  directoryUrl: URL;
  /**
   * Access service URL: URL to the Access Service.
   *
   * @type {URL}
   * @memberof PlatformOptions
   */
  accessServiceUrl: URL;
  /**
   * Ignore service audience: Skip verification of Service ID in token audience.
   *
   * This is generally not necessary. In most cases, use an audience mapper to include the required
   * audience in the access token.
   *
   * @type {boolean}
   * @memberof PlatformOptions
   */
  ignoreServiceAud?: boolean;
  /**
   * Configuration converter: Converter function for configuration; converted value is cached.
   *
   * @type {ConfigurationConverter}
   * @memberof PlatformOptions
   */
  configurationConverter?: ConfigurationConverter;
  /**
   * Combine configuration: Combine function for merging tenant and core configuration.
   *
   * Configuration can be stored in both a tenant specific context and a core (cross-tenant) context.
   * Provide a function to merge configuration retrieved in the two contexts. For example, tenant
   * configuration could override parts of the core configuration for the effective configuration.
   *
   * @type {CombineConfiguration}
   * @memberof PlatformOptions
   */
  combineConfiguration?: CombineConfiguration;
  /**
   * Enable configuration invalidation: Flag indicating if websocket connection should be used to
   * invalidate configuration cache on configuration updates.
   *
   * @type {boolean}
   * @memberof PlatformOptions
   */
  enableConfigurationInvalidation?: boolean;
}

export interface PlatformServices {
  /**
   * Directory: Directory of services.
   *
   * @type {ServiceDirectory}
   * @memberof PlatformServices
   */
  directory: ServiceDirectory;
  /**
   * Tenant service: Service for accessing tenant information.
   *
   * @type {TenantService}
   * @memberof PlatformServices
   */
  tenantService: TenantService;
  /**
   * Configuration service: Service for accessing configuration.
   *
   * @type {ConfigurationService}
   * @memberof PlatformServices
   */
  configurationService: ConfigurationService;
  /**
   * Event service: Service for emitting domain events.
   *
   * @type {EventService}
   * @memberof PlatformServices
   */
  eventService: EventService;
}

export interface PlatformCapabilities extends PlatformServices {
  /**
   * Token provider: Provides service account access token.
   *
   * @type {TokenProvider}
   * @memberof PlatformCapabilities
   */
  tokenProvider: TokenProvider;
  /**
   * Core strategy: PassportJS strategy for core access tokens.
   *
   * Use this strategy to validate access tokens issued by the core realm (cross tenant).
   *
   * @type {Strategy}
   * @memberof PlatformCapabilities
   */
  coreStrategy: Strategy;
  /**
   * Tenant strategy: PassportJS strategy for tenant access tokens.
   *
   * Use this strategy to validate access tokens issued by tenant realms.
   *
   * @type {Strategy}
   * @memberof PlatformCapabilities
   */
  tenantStrategy: Strategy;
  /**
   * Tenant handler: Express middleware that sets req.tenant.
   *
   * @type {RequestHandler}
   * @memberof PlatformCapabilities
   */
  tenantHandler: RequestHandler;
  /**
   * Configuration handler: Express middleware that sets req.configuration.
   *
   * @type {RequestHandler}
   * @memberof PlatformCapabilities
   */
  configurationHandler: RequestHandler;
  /**
   * Health check: Function to retrieve platform health.
   *
   * @type {PlatformHealthCheck}
   * @memberof PlatformCapabilities
   */
  healthCheck: PlatformHealthCheck;
  /**
   * Clear cached: Function to clear the cached configuration.
   *
   * Configuration is cached with a TTL to mitigate stale cache issues. If the service
   * has a mechanism to detect configuration updates, such as a websocket connection,
   * use this function to invalidate the cache.
   *
   * @memberof PlatformCapabilities
   */
  clearCached: (tenantId: AdspId, serviceId: AdspId) => void;
  /**
   * Metrics handler: Request handler that write micro-benches of request times to
   * value service.
   *
   * Note that for platform services metrics are written to tenant contexts. Values are
   * only written if req.tenant is set by the tenantHandler before the response.
   *
   * @type {RequestHandler}
   * @memberof PlatformCapabilities
   */
  metricsHandler: RequestHandler;
  /**
   * Logger used by SDK components.
   *
   * @type {Logger}
   * @memberof PlatformCapabilities
   */
  logger: Logger;
}
