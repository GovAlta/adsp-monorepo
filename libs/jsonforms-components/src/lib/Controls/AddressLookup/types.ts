export interface Address {
  addressLine1: string;
  addressLine2: string;
  city: string;
  province: string;
  postalCode: string;
  country: string;
}

export interface Suggestion {
  Id: string;
  Text: string;
  Highlight: string;
  Cursor: number;
  Description: string;
  Next: string;
}
