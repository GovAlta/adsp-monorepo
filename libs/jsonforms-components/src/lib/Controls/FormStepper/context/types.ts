import { Categorization, Category } from '@jsonforms/core';

export interface CategoryInternalState {
  isCompleted?: boolean;
  isVisited?: boolean;
  isValid?: boolean;
  id: number;
  uischema?: CategorizationElement;
  isEnabled?: boolean;
  visible?: boolean;
  path?: string;
}

export interface CategoryConfig {
  label: string;
  scopes: string[];
}

export type CategoryState = CategoryInternalState & CategoryConfig;

export type CategoriesState = Array<CategoryState>;

export interface StepperContextDataType {
  categories: CategoriesState;
  activeId: number;
  hasNextButton: boolean;
  hasPrevButton: boolean;
  path: string;
  isOnReview: boolean;
  isValid: boolean;
  maxReachedStep: number; // track the max step reached
}

export type CategorizationElement = Category | Categorization;
