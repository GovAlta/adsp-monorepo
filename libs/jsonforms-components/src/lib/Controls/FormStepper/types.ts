import { StatePropsOfLayout } from '@jsonforms/core';
import { AjvProps } from '../../util/layout';
import { TranslateProps } from '@jsonforms/react';
import { ReactNode } from 'react';

export interface CategorizationStepperLayoutRendererProps extends StatePropsOfLayout, AjvProps, TranslateProps {}

export interface CategorizationStepperLayoutReviewRendererProps extends CategorizationStepperLayoutRendererProps {
  navigationFunc?: (index: number) => void;
}

export interface FormStepperComponentProps {
  controlledNav?: number;
  readOnly?: boolean;
}
