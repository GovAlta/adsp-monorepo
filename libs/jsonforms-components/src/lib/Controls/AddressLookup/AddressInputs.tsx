import React from 'react';
import { Address } from './types';
import { GoAFormItem, GoAInput, GoAGrid, GoADropdownItem, GoADropdown } from '@abgov/react-components';
import { LabelDiv } from './styled-components';

interface AddressInputsProps {
  address: Address;
  handleInputChange: (field: string, value: string) => void;
  isAlbertaAddress?: boolean;
  readOnly?: boolean;
  // eslint-disable-next-line
  errors?: any;
  handleOnBlur: (field: string) => void;
  requiredFields?: string[];
  enabled?: boolean;
}

export const AddressInputs: React.FC<AddressInputsProps> = ({
  address,
  handleInputChange,
  isAlbertaAddress,
  readOnly,
  errors,
  handleOnBlur,
  requiredFields,
  enabled,
}: AddressInputsProps): JSX.Element => {
  const provinces = [
    { value: 'AB', label: 'Alberta' },
    { value: 'BC', label: 'British Columbia' },
    { value: 'MB', label: 'Manitoba' },
    { value: 'NB', label: 'New Brunswick' },
    { value: 'NL', label: 'Newfoundland and Labrador' },
    { value: 'NS', label: 'Nova Scotia' },
    { value: 'NT', label: 'Northwest Territories' },
    { value: 'NU', label: 'Nunavut' },
    { value: 'ON', label: 'Ontario' },
    { value: 'PE', label: 'Prince Edward Island' },
    { value: 'QC', label: 'Quebec' },
    { value: 'SK', label: 'Saskatchewan' },
    { value: 'YT', label: 'Yukon' },
  ];

  return (
    <>
      <GoAFormItem label="">
        <GoAInput
          name="addressLine2"
          testId="address-form-address2"
          disabled={!enabled}
          readonly={readOnly}
          ariaLabel={'address-form-address2'}
          placeholder="Unit number, suite, apartment"
          value={address?.addressLine2 || ''}
          onChange={(name, value) => handleInputChange(name, value)}
          width="100%"
        />
      </GoAFormItem>
      <br />
      <GoAGrid minChildWidth="0ch" gap="s">
        <GoAFormItem
          label="City"
          error={errors?.['municipality'] ?? ''}
          requirement={requiredFields?.includes('municipality') ? 'required' : 'optional'}
        >
          <GoAInput
            name="municipality"
            testId="address-form-city"
            ariaLabel={'address-form-city'}
            disabled={!enabled}
            readonly={readOnly}
            value={address?.municipality || ''}
            onChange={(name, value) => handleInputChange(name, value)}
            onBlur={(name) => handleOnBlur(name)}
            width="100%"
          />
        </GoAFormItem>
        <GoAFormItem
          label="Postal code"
          error={errors?.postalCode ?? ''}
          requirement={requiredFields?.includes('postalCode') ? 'required' : 'optional'}
        >
          <GoAInput
            name="postalCode"
            testId="address-form-postal-code"
            ariaLabel={'address-form-postal-code'}
            disabled={!enabled}
            readonly={readOnly}
            placeholder="A0A 0A0"
            value={address?.postalCode || ''}
            onChange={(name, value) => handleInputChange(name, value)}
            onBlur={(name) => handleOnBlur(name)}
            width="100%"
            maxLength={7}
          />
        </GoAFormItem>
      </GoAGrid>
      <br />
      <GoAGrid minChildWidth="0" gap="s">
        <GoAFormItem label="Province">
          {isAlbertaAddress && <LabelDiv data-testid="address-form-province">Alberta</LabelDiv>}
          {!isAlbertaAddress && (
            <GoADropdown
              name="subdivisionCode"
              disabled={!enabled}
              testId="address-form-province-dropdown"
              ariaLabel={'address-form-province'}
              value={address?.subdivisionCode || ''}
              onChange={(_, value) => handleInputChange('subdivisionCode', value as string)}
              relative={true}
              width="25ch"
            >
              {provinces.map((w) => (
                <GoADropdownItem key={w.value} value={w.value} label={w.label} />
              ))}
            </GoADropdown>
          )}
        </GoAFormItem>
        <GoAFormItem label="Country">
          <LabelDiv data-testid="address-form-country">Canada</LabelDiv>
        </GoAFormItem>
      </GoAGrid>
    </>
  );
};
