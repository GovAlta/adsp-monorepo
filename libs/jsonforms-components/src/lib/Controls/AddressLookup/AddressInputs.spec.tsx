import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { AddressInputs } from './AddressInputs';
import { Address } from './types';
describe('AddressInputs', () => {
  const mockHandleInputChange = jest.fn(() => Promise.resolve());
  const mockHandleInputBlur = jest.fn();
  const defaultAddress: Address = {
    addressLine1: '123 Main St',
    addressLine2: 'Apt 4',
    municipality: 'Edmonton',
    subdivisionCode: 'BC',
    postalCode: 'T5T 1V4',
    country: 'CA',
  };
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders all input fields with the correct initial values', () => {
    const component = render(
      <AddressInputs
        address={defaultAddress}
        handleOnBlur={mockHandleInputBlur}
        handleInputChange={mockHandleInputChange}
        isAlbertaAddress={false}
      />
    );
    const addressLine2 = component.getByTestId('address-form-address2');
    expect((addressLine2 as HTMLInputElement).value).toBe(defaultAddress.addressLine2);
    const city = component.getByTestId('address-form-city');
    expect((city as HTMLInputElement).value).toBe(defaultAddress.municipality);
    const province = component.getByTestId('address-form-province-dropdown');
    expect((province as HTMLInputElement).value).toBe(defaultAddress.subdivisionCode);
    const postalCode = component.getByTestId('address-form-postal-code');
    expect((postalCode as HTMLInputElement).value).toBe(defaultAddress.postalCode);
  });

  it('calls handleInputChange on user input in address2', () => {
    render(
      <AddressInputs
        address={defaultAddress}
        handleOnBlur={mockHandleInputBlur}
        handleInputChange={mockHandleInputChange}
      />
    );

    const addressLine2Input = screen.getByTestId('address-form-address2');
    fireEvent.change(addressLine2Input, { target: { value: 'Suite 5' } });

    fireEvent(
      addressLine2Input,
      new CustomEvent('_change', {
        detail: { name: 'addressLine2', value: 'Suite 5' },
      })
    );
    expect((addressLine2Input as HTMLInputElement).value).toBe('Suite 5');
    expect(mockHandleInputChange).toBeCalledTimes(1);
    expect(mockHandleInputChange).toHaveBeenCalledWith('addressLine2', 'Suite 5');
  });
  it('calls handleInputChange on user input in city', () => {
    render(
      <AddressInputs
        address={defaultAddress}
        handleInputChange={mockHandleInputChange}
        handleOnBlur={mockHandleInputBlur}
      />
    );

    const cityInput = screen.getByTestId('address-form-city');
    fireEvent.change(cityInput, { target: { value: 'Calgary' } });
    fireEvent(
      cityInput,
      new CustomEvent('_change', {
        detail: { name: 'city', value: 'Calgary' },
      })
    );
    expect((cityInput as HTMLInputElement).value).toBe('Calgary');
    expect(mockHandleInputChange).toBeCalledTimes(1);
    expect(mockHandleInputChange).toHaveBeenCalledWith('city', 'Calgary');
    const blurred = fireEvent.blur(cityInput);

    expect(blurred).toBe(true);
  });
  it('calls handleInputChange on user input in postal code', () => {
    render(
      <AddressInputs
        address={defaultAddress}
        handleOnBlur={mockHandleInputBlur}
        handleInputChange={mockHandleInputChange}
      />
    );

    const pcInput = screen.getByTestId('address-form-postal-code');
    fireEvent.change(pcInput, { target: { value: 'T2X 2N0' } });
    fireEvent(
      pcInput,
      new CustomEvent('_change', {
        detail: { name: 'postalCode', value: 'T2X 2N0' },
      })
    );
    expect((pcInput as HTMLInputElement).value).toBe('T2X 2N0');
    expect(mockHandleInputChange).toBeCalledTimes(1);
    expect(mockHandleInputChange).toHaveBeenCalledWith('postalCode', 'T2X 2N0');
    const blurred = fireEvent.blur(pcInput);
    expect(blurred).toBe(true);
  });

  it('calls handleInputChange on user input in province for canadian address', () => {
    render(
      <AddressInputs
        address={defaultAddress}
        handleOnBlur={mockHandleInputBlur}
        handleInputChange={mockHandleInputChange}
        isAlbertaAddress={false}
      />
    );

    const provinceInput = screen.getByTestId('address-form-province-dropdown');
    fireEvent.change(provinceInput, { target: { value: 'BC' } });
    fireEvent(
      provinceInput,
      new CustomEvent('_change', {
        detail: { name: 'province', value: 'BC' },
      })
    );

    expect((provinceInput as HTMLInputElement).value).toBe('BC');
    expect(mockHandleInputChange).toBeCalledTimes(1);
    expect(provinceInput).toBeTruthy();
  });
  it(' province is label when isAlbertaAddress is true', () => {
    const component = render(
      <AddressInputs
        address={defaultAddress}
        handleOnBlur={mockHandleInputBlur}
        handleInputChange={mockHandleInputChange}
        isAlbertaAddress={true}
      />
    );

    const provinceInput = component.getByTestId('address-form-province');
    expect(provinceInput).toBeTruthy();
  });
  it('renders default country as Canada ', () => {
    const component = render(
      <AddressInputs
        address={defaultAddress}
        handleOnBlur={mockHandleInputBlur}
        handleInputChange={mockHandleInputChange}
        isAlbertaAddress={true}
      />
    );

    const input = component.getByTestId('address-form-country');

    expect(input).toBeTruthy();
  });
});
