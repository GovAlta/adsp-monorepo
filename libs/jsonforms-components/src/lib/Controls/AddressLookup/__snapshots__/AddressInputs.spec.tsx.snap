// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressInputs matches snapshot 1`] = `
<DocumentFragment>
  <goa-form-item
    label=""
  >
    <goa-input
      arialabel="address-form-address2"
      data-testid="address-form-address2"
      handletrailingiconclick="false"
      name="addressLine2"
      placeholder="Unit number, suite, apartment"
      value="Apt 4"
      variant="goa"
      width="100%"
    />
  </goa-form-item>
  <br />
  <goa-grid
    gap="s"
    minchildwidth="0ch"
  >
    <goa-form-item
      error=""
      label="City"
      requirement="optional"
    >
      <goa-input
        arialabel="address-form-city"
        data-testid="address-form-city"
        handletrailingiconclick="false"
        name="municipality"
        value="Edmonton"
        variant="goa"
        width="100%"
      />
    </goa-form-item>
    <goa-form-item
      error=""
      label="Postal Code"
      requirement="optional"
    >
      <goa-input
        arialabel="address-form-postal-code"
        data-testid="address-form-postal-code"
        handletrailingiconclick="false"
        maxlength="7"
        name="postalCode"
        placeholder="A0A 0A0"
        value="T5T 1V4"
        variant="goa"
        width="100%"
      />
    </goa-form-item>
  </goa-grid>
  <br />
  <goa-grid
    gap="s"
    minchildwidth="0"
  >
    <goa-form-item
      label="Province"
    >

      <div
        class="sc-gswNZR cDxa-dW"
        data-testid="address-form-province"
      >
        Alberta
      </div>
    </goa-form-item>
    <goa-form-item
      label="Country"
    >
      <div
        class="sc-gswNZR cDxa-dW"
        data-testid="address-form-country"
      >
        Canada
      </div>
    </goa-form-item>
  </goa-grid>
</DocumentFragment>
`;
